app.ts

```diff
@@ -1,54 +1,80 @@
+//app.ts
+
 import express, { Request, Response, NextFunction } from 'express';
-import mongoose, { ConnectOptions } from 'mongoose';
-import { routes } from './routes/routes';
-import jwt from 'jsonwebtoken';
+import jwt, { VerifyErrors } from 'jsonwebtoken';
+import { User } from './models/models';
+import { usersUtil } from './utils/utils';
+import { UsersController } from './controller/controller';
+
+const port = 4000;
+const secretKey = 'tu-clave-secreta'; // Reemplaza esto con tu clave secreta para JWT
 
 const app = express();
-const port = 3000;
-const SECRET_KEY = 'gauss626';
-const MONGO_URL = 'mongodb://127.0.0.1:27017/mydatabase';
-const COLLECTION_NAME = 'log-collection';
 
-app.use(express.json());
+// Middleware para verificar el token JWT en las rutas protegidas
+const authenticate = (req: Request, res: Response, next: NextFunction) => {
```

Comment:
This should also validate that the token has `userId` and `userRole === ADMIN` in the token.